src/details/ArborX_DetailsCartesianGrid.hpp

```diff
@@ -17,10 +17,13 @@
 #include <ArborX_HyperBox.hpp>
 
 #include <Kokkos_Macros.hpp>
+#if KOKKOS_VERSION >= 40200
+#include <Kokkos_Assert.hpp> // KOKKOS_ASSERT
+#else
+#include <Kokkos_Core.hpp> // KOKKOS_ASSERT
```

Comment:
Did you consider including `impl/Kokkos_Error.hpp` instead?  I suspect that would work.

Reply:
Cons:
- `impl/Kokkos_Error.hpp` is not a public header. We shouldn't be a bad example.
- up to `Kokkos` version 4.2.0 (at the very least), there is not much of a chance for an `ArborX` executable to be able to include less than `Kokkos_Core.hpp` anyway.

Pros:
- we only include non-public headers up to a version that we know the file exists.
- we don't rely on `Kokkos_Core.hpp` including headers we should provide ourselves, see https://github.com/arborx/ArborX/pull/956, before version 4.2.0.

Reply:
> Cons:
> 
> * `impl/Kokkos_Error.hpp` is not a public header. We shouldn't be a bad example.

In our implementation details only.  I don't think that is an issue.

> * up to `Kokkos` version 4.2.0 (at the very least), there is not much of a chance for an `ArborX` executable to be able to include less than `Kokkos_Core.hpp` anyway.

I don't understand the point you want to make.